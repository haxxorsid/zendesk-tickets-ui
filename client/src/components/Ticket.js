import React from 'react';
import { formattedDate } from '../common/helper';

function Ticket(props) {
    return (
        <div className="ticket" onClick={() => props.onSelectTicket(props.ticket.id)}>
            <div className="ticket-header">
                <p>
                    <span className="id"> ID: {props.ticket.id} </span>
                    <span className="status">Status: {props.ticket.status}</span>
                    <span className="right date">{formattedDate(props.ticket.created_at)}</span>
                </p>
                <p>
                    <b>Subject: </b> <span className="subject">{props.ticket.subject}</span>
                </p>
            </div>
        </div>
    );
}

export default Ticket;