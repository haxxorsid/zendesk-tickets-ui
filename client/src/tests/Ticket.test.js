import React from 'react';
import renderer from 'react-test-renderer'
import Ticket from '../components/Ticket';
import ticketData from './mockData/ticket'

test('Should render ticket row correctly with props', () => {
    const component = renderer
        .create(<Ticket key={ticketData.id} ticket={ticketData} />);
    expect(component.toJSON()).toMatchSnapshot()
  })