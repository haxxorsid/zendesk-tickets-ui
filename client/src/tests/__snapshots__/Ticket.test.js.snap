// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render ticket row correctly with props 1`] = `
<div
  className="ticket"
  onClick={[Function]}
>
  <div
    className="ticket-header"
  >
    <p>
      <span
        className="id"
      >
         ID: 
        2
         
      </span>
      <span
        className="status"
      >
        Status: 
        open
      </span>
      <span
        className="right date"
      >
        Tue Nov 23 2021
      </span>
    </p>
    <p>
      <b>
        Subject: 
      </b>
       
      <span
        className="subject"
      >
        velit eiusmod reprehenderit officia cupidatat
      </span>
    </p>
  </div>
</div>
`;
